src/lib/pubkey/eckcdsa/eckcdsa.cpp

```diff
@@ -72,13 +72,13 @@ std::unique_ptr<HashFunction> eckcdsa_signature_hash(const AlgorithmIdentifier&
    return HashFunction::create_or_throw(oid_info[1]);
 }
 
-std::vector<uint8_t> eckcdsa_prefix(const PointGFp& point, size_t order_bytes, size_t hash_block_size) {
-   const BigInt public_x = point.get_affine_x();
-   const BigInt public_y = point.get_affine_y();
+std::vector<uint8_t> eckcdsa_prefix(const EC_Point& point, size_t hash_block_size) {
+   const auto public_x = point.x_bytes();
+   const auto public_y = point.y_bytes();
 
-   std::vector<uint8_t> prefix(2 * order_bytes);
-   BigInt::encode_1363(&prefix[0], order_bytes, public_x);
-   BigInt::encode_1363(&prefix[order_bytes], order_bytes, public_y);
+   std::vector<uint8_t> prefix(public_x.size() + public_y.size());
+   copy_mem(&prefix[0], public_x.data(), public_x.size());
+   copy_mem(&prefix[public_x.size()], public_y.data(), public_y.size());
```

Comment:
All this could be replaced by a `concat`:

```C++
auto prefix = concat<std::vector<uint8_t>>(point.x_bytes(), point.y_bytes());
```

Reply:
:+1: